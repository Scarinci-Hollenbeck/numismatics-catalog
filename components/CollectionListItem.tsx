import Link from 'next/link';
import { CollectionLink } from '../interfaces';

type Props = {
  collection: CollectionLink
}

export default function CollectionListItem({ collection }: Props): JSX.Element {
  return (
    <li className="collection-item">
      <Link href={`/collection/${encodeURIComponent(collection.id)}/${encodeURIComponent(collection.link)}`}>
      <a>
        {collection.title}
      </a>
    </Link>
    </li>
  )
}