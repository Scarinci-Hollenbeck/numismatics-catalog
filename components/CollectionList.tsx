import react from 'react';
import { CollectionLink } from '../interfaces';
import CollectionListItem from './CollectionListItem';

type Props = {
  collections: CollectionLink[]
}

export default function CollectionList({ collections }): JSX.Element {
  return (
    <section className="collections">
      <h2>List of coin collections</h2>
      <hr />
      <ul className="list">
        {collections.map((collection) => <CollectionListItem key={collection.id} collection={collection} />)}
      </ul>
    </section>
  )
}