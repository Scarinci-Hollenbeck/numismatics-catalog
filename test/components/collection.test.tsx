import React from 'react'
import renderer from 'react-test-renderer'
import CollectionListItem from '../../components/CollectionListItem'
import CollectionList from '../../components/CollectionList'

const collections = [
  {
    id: '1',
    title: 'Calico Kings and Queens of Spain',
    link: 'calico-kings-and-queens-of-spain',
  },
  {
    id: '2',
    title: 'Calico Kings and Queens of Spain',
    link: 'calico-kings-and-queens-of-spain',
  },
  {
    id: '3',
    title: 'Calico Kings and Queens of Spain',
    link: 'calico-kings-and-queens-of-spain',
  },
]

describe('Collection list Item component', () => {
  it('matches snapshot', () => {
    const tree = renderer
      .create(<CollectionListItem collection={collections[0]} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})

describe('Collection list Container component', () => {
  it('matches snapshot', () => {
    const tree = renderer
      .create(<CollectionList collections={collections} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
