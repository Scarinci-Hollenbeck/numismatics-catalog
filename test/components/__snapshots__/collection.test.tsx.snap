// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collection list Container component matches snapshot 1`] = `
<section
  className="collections"
>
  <h2>
    List of coin collections
  </h2>
  <hr />
  <ul
    className="list"
  >
    <li
      className="collection-item"
    >
      <a
        href="/collection/1/calico-kings-and-queens-of-spain"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Calico Kings and Queens of Spain
      </a>
    </li>
    <li
      className="collection-item"
    >
      <a
        href="/collection/2/calico-kings-and-queens-of-spain"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Calico Kings and Queens of Spain
      </a>
    </li>
    <li
      className="collection-item"
    >
      <a
        href="/collection/3/calico-kings-and-queens-of-spain"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Calico Kings and Queens of Spain
      </a>
    </li>
  </ul>
</section>
`;

exports[`Collection list Item component matches snapshot 1`] = `
<li
  className="collection-item"
>
  <a
    href="/collection/1/calico-kings-and-queens-of-spain"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    Calico Kings and Queens of Spain
  </a>
</li>
`;
